docker/docker_image_src.go

```diff
@@ -377,7 +379,7 @@ func (s *dockerImageSource) getOneSignature(ctx context.Context, url *url.URL) (
 		} else if res.StatusCode != http.StatusOK {
 			return nil, false, errors.Errorf("Error reading signature from %s: status %d (%s)", url.String(), res.StatusCode, http.StatusText(res.StatusCode))
 		}
-		sig, err := ioutil.ReadAll(res.Body)
+		sig, err := util.ReadAtMost(res.Body, 4*util.MegaByte)
```

Comment:
I recommend to re-declare two consts which describe their purpose.

Reply:
I had that initially and didn't like it, as we needed more than two consts. Basically for each individual blob (manifest, config, authtoken, errorresponse size, signature, signature list, etc.). I much prefer to specify the limit in each call and use the generic `util.MegaByte`.

Reply:
Non-blocking, I’d also prefer a named constant — to make it clear that this value is an arbitrary choice, not something protocol-defined that has to be hard-coded to that exact value.

`notBigMetadataSize` or something like that (covering all of errors, manifests, configs, signatures,…) would still be clearer in purpose than `MegaByte`.

OTOH that can’t be a public API of the util package, it would have to be a separate constant in each package.

Reply:
Everything's in `internal/iolimits` now with individual limits.